bgpd/bgp_route.c

```diff
@@ -4857,7 +4857,7 @@ void bgp_update(struct peer *peer, const struct prefix *p, uint32_t addpath_id,
 		bgp_unlink_nexthop(new);
 		bgp_path_info_delete(dest, new);
 	}
-
+        bgp_dest_unlock_node(dest);
```

Comment:
This is wrong. dest is already unlocked above, and here you are trying to delete `dest` at all.